libmscore/xmlreader.cpp

```diff
@@ -563,7 +563,7 @@ void XmlReader::performReadAhead(std::function<void(QIODevice&)> f)
             _readAheadDevice->open(QIODevice::ReadOnly);
 
       const auto pos = _readAheadDevice->pos();
-      _readAheadDevice->seek(characterOffset());
+      _readAheadDevice->seek(0);
```

Comment:
Now the name of the method does not match what it does.
I think you need to name it differently, for example `rereadDevice`
Or better to remove this method (and `setReadAheadDevice`, `readAheadAvailable`, `_readAheadDevice`) from XmlReader, add a separate function to read206.cpp, which will do `device->seek(0);` and return it back `device->seek(pos);`   
`QIODevice` can be obtained `XmlReader::device()` (`QXmlStreamReader::device()`)

Reply:
Ok. I rewrote the functions and implemented a "read style name" function in read206.cpp.
Function QXmlStreamReader::device() cannot be used because for MuseScore XmlReader it returns a null pointer (that's probably the reason for the comment "Ownership on read device is NOT transferred to XmlReader" in xml.h). So I simply renamed the saved QIODevice pointer and added a method to fetch that pointer.